apps/address-service/domains/common/utils/services/suggest/SuggestionKeystoneApp.js

```diff
@@ -18,37 +18,55 @@ class SuggestionKeystoneApp {
             next()
         }
 
-        app.get('/suggest', setNoCache, async (req, res, next) => {
+        app.all('/suggest', setNoCache, async (req, res, next) => {
+            if (!['GET', 'POST'].includes(req.method)) {
+                res.send(404)
+            }
+
+            const reqQuery = get(req, 'query', {})
+            const reqBody = get(req, 'body', {})
+
+            /**
+             * @param {string} param
+             * @param {*} [def]
+             */
+            const getParam = (param, def) => get(reqBody, param, get(reqQuery, param, def))
+
             /**
              * Using to detect a proper suggestion provider
              * @type {string}
              */
-            const geo = String(get(req, ['query', 'geo'], null))
+            const geo = String(getParam('geo', null))
 
             /**
              * User's search string
              * @type {?string}
              */
-            const s = get(req, ['query', 's'])
+            const s = getParam('s')
 
             /**
              * To skip normalization of results. If true - returns raw provider's result
              * @type {boolean}
              */
-            const bypass = Boolean(get(req, ['query', 'bypass'], false))
+            const bypass = Boolean(getParam('bypass', false))
 
             /**
              * Sometimes we need to use different query parameters to providers
              * depending on different clients (mobile app, backend job, user runtime)
              * @type {string}
              */
-            const context = String(get(req, ['query', 'context'], ''))
+            const context = String(getParam('context', ''))
 
             /**
              * Number of results to return
              * @type {number|NaN}
              */
-            const count = Number(get(req, ['query', 'count'], undefined))
+            const count = Number(getParam('count'))
```

Comment:
Number(undefined) -> NaN

Reply:
Yes, it's ok. I want to get NaN in the case of no parameter. The default value already set within particular provider.